Firestore/core/src/firebase/firestore/util/status.h

```diff
@@ -0,0 +1,95 @@
+// TODO(rsgowman): how should the copyright be adapted?
+// Protocol Buffers - Google's data interchange format
+// Copyright 2008 Google Inc.  All rights reserved.
+// https://developers.google.com/protocol-buffers/
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are
+// met:
+//
+//     * Redistributions of source code must retain the above copyright
+// notice, this list of conditions and the following disclaimer.
+//     * Redistributions in binary form must reproduce the above
+// copyright notice, this list of conditions and the following disclaimer
+// in the documentation and/or other materials provided with the
+// distribution.
+//     * Neither the name of Google Inc. nor the names of its
+// contributors may be used to endorse or promote products derived from
+// this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+// "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+// LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+// A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+// OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+// SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+// LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+// DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+// THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+// (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+// OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+#ifndef FIRESTORE_CORE_SRC_FIREBASE_FIRESTORE_UTIL_STATUS_H_
+#define FIRESTORE_CORE_SRC_FIREBASE_FIRESTORE_UTIL_STATUS_H_
+
+#include <iosfwd>
+#include <string>
+
+#include "Firestore/core/include/firebase/firestore/firestore_errors.h"
+#include "absl/strings/string_view.h"
+
+namespace firebase {
+namespace firestore {
+namespace util {
+
+class Status {
+ public:
+  // Creates a "successful" status.
+  Status();
+
+  // Create a status in the with the specified code, and error message.  If
+  // "code == 0", error_message is ignored and a Status object identical to
+  // Status::OK is constructed.
+  Status(FirestoreErrorCode error_code, absl::string_view error_message);
+  Status(const Status&);
+  Status& operator=(const Status& x);
+  ~Status() {
+  }
+
+  // Some pre-defined Status objects
+  static const Status OK;  // Identical to 0-arg constructor
```

Comment:
The constant naming convention is incorrect. But ignore me if it is named so to match something we want to replace to in the future.

Reply:
It's to match the existing status.h implementations. I'm not sure why the original authors diverged; possibly because Status:kOk looks awkward or maybe to match up with the proto enums (http://google3/util/task/codes.proto).

This PR is about to be abandoned, but the same issue will arise on the tensorflow version which is about to be sent out (since tf also uses Status::OK()... though as a static factory method rather than as a constant. Still violates the style guide though.)